File: PerfTester-Xamarin/PerfTester-Xamarin/Models/HeavyCalculationsSyncTest.cs
using System;
using System.Collections.Generic;

namespace PerfTesterXamarin.Models
{
    public class HeavyCalculationsSyncTest : Test
    {
        public override string Title => "Heavy Sync Calculations";
        public override string Desc => "Doing a lot of math operations on a huge matrixes working on a main thread.";
        public override string ImageName => "Maths";
        public override double[] Parameters => new double[] { 10000, 20000, 50000, 100000, 200000, 500000, 1000000, 2000000, 5000000, 10000000 };

        public override void Prepare(double param)
        {}

        public override void DoJob(double param)
        {
            for (long i = 0; i < param; i++)
            {
                //List<double> testArr = new List<double> { 32.521, 643.321, 74532.21, 634.12, 0.042021, -532.321, -321.6732, -2213.32535 };
                //List<double> finArr = MultiplyArray(testArr);
            }
            base.FinishJob(param);
        }

        List<double> MultiplyArray(List<double> array)
        {
            List<double> finArr = new List<double>();
            for (int i = 0; i < array.Count; i++)
            {
                finArr.Add(array[i] * array[i]);
            }
            return finArr;
        }
    }
}

File: PerfTester-Xcode/PerfTester-Xcode/Models/HeavyCalculationsSyncTest.swift
//
//  HeavyCalculationsSyncTest.swift
//  PerfTester-Xcode
//
//  Created by Patryk Średziński on 08.06.2018.
//  Copyright © 2018 private.perftesterxcode. All rights reserved.
//

import UIKit

class HeavyCalculationsSyncTest : Test {

    override init() {
        super.init()
        title = "Heavy Sync Calculations"
        desc = "Creating N times an array of 8 doubles and calculating it's power of 2 on a main thread"
        imageName = "Maths"
        parameters = [10000, 20000, 50000, 100000, 200000, 500000, 1000000, 2000000, 5000000, 10000000]
    }
    
    override func prepare(param: Double) {
        
    }
    
    override func done(param: Double) {

    }
    
    override func doJob(param: Double) {
        for _ in 1...Int(param) {
            let testArr: [Double] = [32.521, 643.321, 74532.21, 634.12, 0.042021, -532.321, -321.6732, -2213.32535]
            var _ = multiplyArray(testArr)
        }
        super.finishJob(param: param)
    }
    
    private func multiplyArray(_ array: [Double]) -> [Double] {
        var finArr = [Double](repeating: 0, count: array.count)
        for i in 1...array.count-1 {
            finArr[i] = array[i] * array[i]
        }
        return finArr
    }
    
}
